web/app.fluidity.money/app/routes/$network/dashboard/airdrop/index.tsx

```diff
@@ -289,20 +305,131 @@ const Airdrop = () => {
     };
 
     const YouAreEligible = () => {
+      // used to track if the button was clicked to delegate. not disabled.
+      // the user will have to reload the page to set it again. unless an error happened!
+      const [beginAssociation, setBeginAssociation] = useState(false);
+
+      // used to paint the button backdrop to set the disabled
+      const [validAssociationAddress, setValidAssociationAddress] = useState(false);
+
+      // used to prevent double submitting until the request is done
+      const [airdropAssociationInProgress, setAirdropAssociationInProgress] = useState(false);
+
+      // the ethereum address they want to associate
+      const [associationEthereumAddress, setAssociationEthereumAddress] = useState("");
+
+      const [hasEthereumAddressChangedSinceClick, setHasEthereumAddressChangedSinceClick] =
+        useState(true);
+
+      // association worked, make the button green and disable the button
+      const [airdropAssociationSucceeded, setAirdropAssociationSucceeded] = useState(false);
+
+      const handleSetAssociationEthereumAddress = (s: string) => {
+        setAssociationEthereumAddress(s);
+        setHasEthereumAddressChangedSinceClick(true);
+        setEthereumAddressHint("");
+      };
+
+      const [ethereumAddressHint, setEthereumAddressHint] = useState("");
+
+      useEffect(() => {
+        (async () => {
+          if (!beginAssociation)
+            return;
+          try {
+            const sig = await airdropAssociateEthereumAccount!(associationEthereumAddress);
+            console.log("done airdrop association with signature", sig);
+            setAirdropAssociationSucceeded(true);
+            setEthereumAddressHint("Succeeded!");
+          } catch (err) {
+            // we set this to true so we can begin again with this attempt.
+            setBeginAssociation(false);
+            setEthereumAddressHint(`${err}`);
+            console.warn("error submitting claim feature", err);
+          }
+          setAirdropAssociationInProgress(false);
+        })();
+      }, [
+        beginAssociation
+      ]);
+
+      const handleAirdropAssociation = () => {
+        // first, check that the address is okay. then, if it is, begin the
+        // process of using anchor to send out the transaction using
+        // Phantom that associates the address.
+
+        const s = associationEthereumAddress;
+
+        setHasEthereumAddressChangedSinceClick(false);
+
+        // regex taken from the transfer route
+        const addressRegex = /(0x[a-fA-F0-9]{40})/g;
+        const validAddr = s.match(addressRegex) || ethers.utils.isAddress(s);
```

Comment:
why do both of these?

Reply:
I took this from the existing send/receive implementation. I'm just going to assume it's fine for now.